crates/stages/stages/src/stages/sender_recovery.rs

```diff
@@ -673,7 +673,10 @@ mod tests {
                             let transaction: TransactionSigned = provider
                                 .transaction_by_id_unhashed(tx_id)?
                                 .map(|tx| {
-                                    TransactionSigned::new_unhashed(tx.transaction, tx.signature)
+                                    TransactionSigned::new_unhashed(
+                                        tx.transaction().clone(),
+                                        *tx.signature(),
```

Comment:
this looks odd, why does this require a conversion at all,

in any case we need decomposition fns like

https://github.com/paradigmxyz/reth/blob/cd44fc3d093c191128bf7bd9fc90b5231d2c1ab2/crates/primitives-traits/src/block/mod.rs#L99-L100